tests/GrabyTest.php

```diff
@@ -418,24 +419,33 @@ public function testAssetExtensionTXT()
     public function dataForSinglePage()
     {
         return [
-            // single_page_link will return a string (ie the text content of <a> node)
-            ['singlepage1.com', 'http://singlepage1.com/printed view', 'http://moreintelligentlife.com/print/content'],
-            // single_page_link will return the a node
-            ['singlepage2.com', 'http://singlepage2.com/print/content', 'http://singlepage2.com/print/content'],
-            // single_page_link will return the href from a node
-            ['singlepage3.com', 'http://singlepage3.com/print/content', 'http://singlepage3.com/print/content'],
-            // single_page_link will return nothing useful
-            ['singlepage4.com', 'http://singlepage4.com', 'http://singlepage4.com/print/content'],
-            // single_page_link will return the href from a node BUT the single page url will be the same
-            ['singlepage3.com/print/content', 'http://singlepage3.com/print/content', 'http://singlepage3.com/print/content'],
+            'single_page_link will return a string (ie the text content of <a> node)' =>
+                ['singlepage1.com', 'http://singlepage1.com/printed view', 'http://moreintelligentlife.com/print/content'],
+            'single_page_link will return the a node' =>
+                ['singlepage2.com', 'http://singlepage2.com/print/content', 'http://singlepage2.com/print/content'],
+            'single_page_link will return the href from a node' =>
+                ['singlepage3.com', 'http://singlepage3.com/print/content', 'http://singlepage3.com/print/content'],
+            'single_page_link will return nothing useful' =>
+                ['singlepage4.com', 'http://singlepage4.com', 'http://singlepage4.com/print/content'],
+            'single_page_link will return the href from a node BUT the single page url will be the same' =>
+                ['singlepage3.com/print/content', 'http://singlepage3.com/print/content', 'http://singlepage3.com/print/content'],
         ];
     }
 
     /**
+     * @group dns-sensitive
      * @dataProvider dataForSinglePage
      */
     public function testSinglePage($url, $expectedUrl, $singlePageUrl)
     {
+        DnsMock::withMockedHosts([
+            'singlepage1.com' => [['type' => 'A', 'ip' => '93.184.216.34']],
```

Comment:
Is it some random IP?

Reply:
IP from example.com

Reply:
Maybe you should put it in a constant to avoid XX updates if example.com changes its dns

Reply:
Others are random ip. `93.184.216.34`  is `example.com`. The test use a http mock client, so a change doesn't impact the test. I will add a constant for the IP, it improves readability